pkg/service/metadata/frontgate/server.go

```diff
@@ -97,7 +97,12 @@ func ServeReverseRpcServerForPilot(
 			lastErrCode = codes.OK
 		}
 
+		connClosed := make(chan struct{})
+		logger.Info(nil, "Starting updater")
+		updater := NewUpdater(conn, connClosed, cfg)
```

Comment:
I think mv this to `func Serve(cfg *ConfigManager, tlsPilotConfig *tls.Config) {` will be better

Reply:
I think it's not a good idea, because updater need a connection to interact with pilot, and when this connection is closed, updater must quit subsequently, then a new connection is established, a new updater will be running on the new connection.